HHH-15216 Cannot change MetadataProvider implementation because JPAXMLOverriddenMetadataProvider is final and precisely expected by a cast operator

reda-alaoui opened the pull request:
https://hibernate.atlassian.net/browse/HHH-15216

This PR moves `JPAXMLOverriddenMetadataProvider` out of package `internal` and makes it non final.

reda-alaoui commented:
Hello?

beikov commented:
Please rebase the PR and make sure you annotate the classes which you move with `@Internal`.

beikov reviewed (commented):
Can you explain why you had to move `PropertyMappingElementCollector`? Also, please squash the commits into a single one.

reda-alaoui commented:
Hello @beikov ,

> Please rebase the PR and make sure you annotate the classes which you move with @Internal.

This is done.

> Can you explain why you had to move PropertyMappingElementCollector?

I wanted to avoid changing its visibility to public. But as you can see now, I reverted the move and changed its visibility to public.

beikov commented:
>But as you can see now, I reverted the move and changed its visibility to public.

But why do you need this in the first place? The class is only used inside of `JPAXMLOverriddenAnnotationReader` and you can't interfere with the usage.

Also, there seem to be compilation errors

reda-alaoui commented:
> But why do you need this in the first place? The class is only used inside of JPAXMLOverriddenAnnotationReader and you can't interfere with the usage.

I need to make it public because the class resides in package internal while I moved one of its consumer to the parent package.
`org.hibernate.cfg.annotations.reflection.JPAXMLOverriddenAnnotationReader` can't access `org.hibernate.cfg.annotations.reflection.internal.PropertyMappingElementCollector` otherwise.

reda-alaoui commented:
The compilation failure should be fixed.

beikov commented:
Thanks

reda-alaoui commented:
Thank you!
Could this be back ported to 5.6.x?

beikov commented:
If you provide a PR for it, then yeah.

reda-alaoui commented:
Ok I will asap 👍
